src/Xml/Reader/MatchingNode.php

```diff
@@ -0,0 +1,39 @@
+<?php
+declare(strict_types=1);
+
+namespace VeeWee\Xml\Reader;
+
+use DOMDocument;
+use VeeWee\Xml\Dom\Document;
+use VeeWee\Xml\Exception\RuntimeException;
+use VeeWee\Xml\Reader\Node\NodeSequence;
+
+final class MatchingNode
+{
+    /**
+     * @param non-empty-string $xml
+     */
+    public function __construct(
+        public readonly string $xml,
```

Comment:
maybe better to provide private props +  getters to keep those DTOs consistent throughout the codebase.